titiler/api/endpoints/mosaic.py

```diff
@@ -0,0 +1,401 @@
+    # available in normal ``def`` functions.
+    # https://github.com/cogeotiff/rio-tiler-mosaic/blob/master/rio_tiler_mosaic/mosaic.py#L37-L102
+    _tiler = partial(
+        cogeoTiler,
+        tile_x=x,
+        tile_y=y,
+        tile_z=z,
+        tilesize=tilesize,
+        indexes=image_params.indexes,
+        # expression=image_params.expression, # TODO: Figure out why expression kwarg doesn't work
+        nodata=image_params.nodata,
+        **image_params.kwargs,
+    )
+    futures = [run_in_threadpool(_tiler, asset) for asset in assets]
+
+    semaphore = asyncio.Semaphore(int(os.getenv("MOSAIC_CONCURRENCY", 10)))
+
+    with utils.Timer() as t:
+        async with semaphore:
+            for fut in asyncio.as_completed(futures):
+                try:
+                    tile, mask = await fut
+                except Exception:
+                    # Gracefully handle exceptions
+                    continue
+
+                tile = numpy.ma.array(tile)
+                tile.mask = mask == 0
+                pixsel.feed(tile)
+                if pixsel.is_done:
+                    break
+    timings.append(("Read-tiles", t.elapsed))
+
+    tile, mask = pixsel.data
+    if tile is None:
+        raise TileNotFoundError(f"Tile {z}/{x}/{y} was not found")
+
+    if not format:
+        format = ImageType.jpg if mask.all() else ImageType.png
+
+    with utils.Timer() as t:
+        tile = utils.postprocess(
+            tile,
+            mask,
+            rescale=image_params.rescale,
+            color_formula=image_params.color_formula,
+        )
+    timings.append(("Post-process", t.elapsed))
+
+    bounds = mercantile.xy_bounds(mercantile.Tile(x, y, z))
+    dst_transform = from_bounds(*bounds, tilesize, tilesize)
+
+    with utils.Timer() as t:
+        content = utils.reformat(
+            tile,
+            mask,
+            img_format=format,
+            colormap=image_params.color_map,
+            dst_transform=dst_transform,
+            crs=CRS.from_epsg(3857),
+        )
+    timings.append(("Format", t.elapsed))
+
+    if timings:
+        headers["X-Server-Timings"] = "; ".join(
+            ["{} - {:0.2f}".format(name, time * 1000) for (name, time) in timings]
+        )
+
+    return ImgResponse(
+        content, media_type=ImageMimeTypes[format.value].value, headers=headers
+    )
+
+
+@router.get("/WMTSCapabilities.xml", response_class=XMLResponse, tags=["OGC"])
+def wmts(
+    request: Request,
+    tile_format: ImageType = Query(
+        ImageType.png, description="Output image type. Default is png."
+    ),
+    tile_scale: int = Query(
+        1, gt=0, lt=4, description="Tile size scale. 1=256x256, 2=512x512..."
+    ),
+    mosaic_params: CommonMosaicParams = Depends(),
+):
+    """OGC WMTS endpoint."""
+    scheme = request.url.scheme
+    host = request.headers["host"]
+    endpoint = f"{scheme}://{host}/mosaic"
```

Comment:
maybe we can use `url_for("read_mosaicjson")`?

Reply:
Oh I remember why I didn't use `url_for`, looks like the wmts template takes the base url of the router.

Reply:
https://github.com/arturo-ai/titiler/blob/52b416d61244e3cb02ea7b68cad252a3387de2b2/titiler/api/endpoints/mosaic.py#L76

`url_for("read_mosaicjson")` should return the base path `:endpoint:/mosaic` ?

Reply:
you're very right haha

Reply:
https://github.com/developmentseed/titiler/pull/40/commits/4602df3f26d527c3a5778a093f25d27e339040ff